### PR TITLE
fix: Xcode and SwiftLint warnings across the entire project

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

From **151 warnings** down to **110**

With this PR I fixed a variety of SwiftLint and Xcode warnings across the entire project. Here's the list of the warnings that are fixed on this PR:

- Call to method 'finishWithoutShowingNotification' in closure requires explicit use of 'self' to make capture semantics explicit; this is an error in Swift 6
- Unneeded Overridden Functions Violation: Remove overridden functions that don't do anything except call their super `(unneeded_override)`
- Variable 'rewindedStep' was never mutated; consider changing to 'let' constant
- Trailing Whitespace Violation: Lines should not have trailing whitespace `(trailing_whitespace)`
- String interpolation produces a debug description for an optional value; did you mean to make this explicit?
- Control Statement Violation: `if`, `for`, `guard`, `switch`, `while`, and `catch` statements shouldn't unnecessarily wrap their conditionals or arguments in parentheses `(control_statement)`
- Initialization of immutable value 'recordButtonFrame' was never used; consider replacing with assignment to '_' or removing it
- Left side of nil coalescing operator '??' has non-optional type 'String', so the right side is never used
- Variable 'user' was never mutated; consider changing to 'let' constant
- Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their closing brace `(switch_case_alignment)`
- Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration `(opening_brace)`

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/EstablishingCallStatusView.swift [hunk 1]
```diff
@@ -40,7 +40,7 @@ extension CallStatusViewState {
         switch self {
         case .none: return ""
         case .connecting: return CallStatus.connecting
-        case .ringingIncoming(name: let name?): return CallStatus.Incoming.user("\(name ?? "")")
+        case .ringingIncoming(name: let name?): return CallStatus.Incoming.user("\(name)")
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/EstablishingCallStatusView.swift:43]: Is "name" not optional?

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift [hunk 2]
```diff
@@ -166,7 +166,7 @@ final class ConversationAvatarView: UIView {
                 return
             }
 
-            accessibilityLabel = "Avatar for \(conversation.displayName)"
+            accessibilityLabel = "Avatar for \(conversation.displayNameWithFallback)"
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift:169]: Which warning was this?

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift:169]: String interpolation produces a debug description for an optional value; did you mean to make this explicit?

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/EstablishingCallStatusView.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Calling/BottomSheetContainerViewController/EstablishingCallStatusView.swift:43]: No! In that case I fixed this error:

Left side of nil coalescing operator '??' has non-optional type 'String', so the right side is never used

File: wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift (hunk 2 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift:169]: What's the fallback?

Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift:169]: If we want to keep the original behavior, we can use `String(describing: conversation.displayName)`

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift:169]: @johnxnguyen 

```swift
extension ConversationLike {

    var displayNameWithFallback: String {
        return displayName ?? L10n.Localizable.Profile.Details.Title.unavailable
    }

}
```

The fallback is `Name not Available` as recently introduced by Katerina! I guess we can use that. and that's why I'm using it.

File: wire-ios-data-model/Source/Model/Message/ConversationMessage+Deletion.swift [hunk 3]
```diff
@@ -67,7 +67,7 @@ extension ZMMessage {
     }
 
     @discardableResult @objc func deleteForEveryone() -> ZMClientMessage? {
```
Comment (theswiftway) [inline wire-ios-data-model/Source/Model/Message/ConversationMessage+Deletion.swift:69]: [Suggestion] Should this return a `Result` instead ?

Comment (theswiftway) [reply wire-ios-data-model/Source/Model/Message/ConversationMessage+Deletion.swift:69]: Not in the scope of this PR though

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Text/ConversationQuoteCell.swift [hunk 4]
```diff
@@ -37,12 +37,12 @@ final class ConversationReplyContentView: UIView {
 
         var showDetails: Bool {
             guard let message = quotedMessage,
-                  (message.isText
+                  message.isText
```
Comment (theswiftway) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Text/ConversationQuoteCell.swift:40]: Looking at all the checks, I think `isFile` check is good enough for this logic

File: wire-ios-data-model/Source/Model/Message/ConversationMessage+Deletion.swift (hunk 3 above)
Comment (agisilaos) [reply wire-ios-data-model/Source/Model/Message/ConversationMessage+Deletion.swift:69]: I agree let's explore that on a separate PR.

File: wire-ios-data-model/Source/Model/Message/ConversationMessage.swift [hunk 5]
```diff
@@ -293,7 +293,7 @@ extension ZMMessage: ZMConversationMessage {
 
     public var isRestricted: Bool {
         guard
-            (self.isFile || self.isImage),
+            self.isFile || self.isImage,
```
Comment (theswiftway) [inline wire-ios-data-model/Source/Model/Message/ConversationMessage.swift:296]: `self.isFile` should be good enough. We could skip `self.isImage`

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Text/ConversationQuoteCell.swift (hunk 4 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Content/Text/ConversationQuoteCell.swift:40]: let's explore that on a separate PR. On this one, I would like to focus on fixing the warnings.

File: wire-ios-data-model/Source/Model/Message/ConversationMessage.swift (hunk 5 above)
Comment (agisilaos) [reply wire-ios-data-model/Source/Model/Message/ConversationMessage.swift:296]: let's explore that on a separate PR. On this one, I would like to focus on fixing the warnings.

**Review by theswiftway — COMMENTED** [review]
LGTM

**Review by theswiftway — APPROVED** [review]

**Review by markusfassbender — APPROVED** [review]
Great styling cleanup!

**Pull request merged**